### PR TITLE
Script uploading release assets

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- shell script uploading manifest and default CR to release assets

**Related issue(s)**

### CODE REVIEW

File: scripts/upload_release_assets.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#     - releaseID (mandatory)
+# ./upload_release_assets.sh 12345678
+
+set -x
+
+RELEASE_ID=${1}
+
+# standard bash error handling
+set -o nounset  # treat unset variables as an error and exit immediately.
+set -o errexit  # exit immediately when a command fails.
```
Comment (ralikio) [inline scripts/upload_release_assets.sh:13]: Can we shorten it and use `set -e`?

**Review by ralikio — CHANGES_REQUESTED** [review]

File: scripts/upload_release_assets.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+set -E          # needs to be set if we want the ERR trap
```
Comment (ralikio) [inline scripts/upload_release_assets.sh:14]: Do we use error traps?

File: scripts/upload_release_assets.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+set -o pipefail # prevents errors in a pipeline from being masked
+
+# Expected variables:
+#   PULL_BASE_REF - name of the tag
+#   BOT_GITHUB_TOKEN - github token used to upload the template yaml
+
+uploadFile() {
+  filePath=${1}
```
Comment (ralikio) [inline scripts/upload_release_assets.sh:22]: Are braces required?

File: scripts/upload_release_assets.sh (hunk 1 above)
Comment (jaroslaw-pieszka) [reply scripts/upload_release_assets.sh:13]: We consistently use this code across plethora of other scripts. Do you want to change all the scripts?

File: scripts/upload_release_assets.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+MANIFEST_FILE="./manifests/btp-operator/btp-manager.yaml"
+DEFAULT_CR_FILE="./examples/btp-operator.yaml"
+UPLOAD_URL="https://uploads.github.com/repos/kyma-project/btp-manager/releases/${RELEASE_ID}/assets"
+
+echo -e "\n--- Updating GitHub release ${RELEASE_ID} with btp-manager.yaml and btp-operator-default-cr.yaml assets"
+
+[[ ! -e ${MANIFEST_FILE} ]] && echo "::error ::Manifest file does not exist" && exit 1
+
+uploadFile "${MANIFEST_FILE}" "${UPLOAD_URL}?name=btp-manager.yaml"
```
Comment (ralikio) [inline scripts/upload_release_assets.sh:48]: What is the reason for quotes in `uploadFile "${MANIFEST_FILE}"`?

File: scripts/upload_release_assets.sh (hunk 2 above)
Comment (jaroslaw-pieszka) [reply scripts/upload_release_assets.sh:14]: See above comment.

File: scripts/upload_release_assets.sh (hunk 1 above)
Comment (ralikio) [reply scripts/upload_release_assets.sh:13]: No need to do it now. Let's discuss it after the PR is merged.

File: scripts/upload_release_assets.sh (hunk 3 above)
Comment (ralikio) [reply scripts/upload_release_assets.sh:22]: Conventions used from other scripts.

**Review by ralikio — APPROVED** [review]

File: scripts/upload_release_assets.sh (hunk 4 above)
Comment (jaroslaw-pieszka) [reply scripts/upload_release_assets.sh:48]: To get rid of IDE warning.

**Pull request merged**